src/main/scala/io/qbeast/context/QbeastContext.scala

```diff
@@ -75,7 +75,7 @@ trait QbeastContext {
  */
 object QbeastContext
     extends QbeastContext
-    with QbeastCoreContext[DataFrame, StructType, QbeastOptions, FileAction] {
+    with QbeastCoreContext[DataFrame, StructType, QbeastOptions, IndexFile] {
```

Comment:
If we use the IndexFile for all internal data management, we don't need to have it parametrized, we can simplify the QbeastCoreContext assuming that we always use IndexFiles.

Reply:
Fixed